crates/codegen/parser/runtime/src/cst.rs

```diff
@@ -46,7 +46,8 @@ impl Node {
         }
     }
 
-    pub fn create_cursor(&self, text_offset: TextIndex) -> Cursor {
+    /// Creates a [`Cursor`] that starts at the current node as the root and a given initial `text_offset`.
+    pub fn cursor_with_offset(&self, text_offset: TextIndex) -> Cursor {
```

Comment:
not sure I understand why we are renaming `create_cursor()` to `cursor_with_offset()`.

The distinct `create` verb indicates that the API creates a new cursor every time, instead of returning a property of `self`. Thoughts on using something like `create_cursor_at()`.

Reply:
I made the change wanting to better convey what does the argument mean at the call-site and what does the `TextIndex` mean in that case - it's an initial offset for the new cursor at that node (rather than, let's say, creating a cursor to a descendent node that contains the given `TextIndex` or similar).

I understand the common API naming guidelines to use verbs/verb phrases but `create` here focuses on a detail that's mostly irrelevant (most of the functions that return new type already do "create", in a sense; and the creation logic here is not complex or costly) while not focusing on the important bit (offset); also these type-to-their-general-iterator-type functions are often named concisely in Rust, e.g. ubiquitous `iter`, `HashMap::keys/values`, `Path::components`, `proc_macro::Group::stream` etc. 

I think the `self`/properties distinction would be more important in an object-oriented paradigm (`createCursor` sounds like a good fit for an object in TS!) but as dealing with data, whether creating, modifying or dropping, directly is a lot more commonplace in Rust, I feel that `create` doesn't add as much value here as explaining the offset, for example (and also makes sense if we decide to add another absolute cursor function in the future).

Reply:
Alright. Thanks for explaining.